web_src/js/features/tribute.js

```diff
@@ -0,0 +1,61 @@
+import {emojiKeys, emojiHTML} from './emoji.js';
+
+export const issuesTribute = window.config.Tribute ? new Tribute({
+  values: window.config.tributeValues,
+  noMatchTemplate() { return null },
+  menuItemTemplate(item) {
+    const div = $('<div/>');
+    div.append($('<img/>', {src: item.original.avatar}));
+    div.append($('<span/>', {class: 'name'}).text(item.original.name));
+    if (item.original.fullname && item.original.fullname !== '') {
+      div.append($('<span/>', {class: 'fullname'}).text(item.original.fullname));
+    }
+    return div.html();
+  }
+}) : null;
+
+export const emojiTribute = window.config.Tribute ? new Tribute({
+  collection: [{
+    trigger: ':',
+    requireLeadingSpace: true,
+    values(query, cb) {
+      const matches = [];
+      for (const name of emojiKeys) {
+        if (name.includes(query)) {
+          matches.push(name);
+          if (matches.length > 5) break;
+        }
+      }
+      cb(matches);
+    },
+    lookup(item) {
+      return item;
+    },
+    selectTemplate(item) {
+      if (typeof item === 'undefined') return null;
+      return `:${item.original}:`;
```

Comment:
@silverwind sorry not sure when this was added but this should insert the actual emoji character as it did before and not the short code. Lots of these changes are breaking or changing what was existing intentional and desired behavior which wasn't really my intention when asking for help making the javascript better and not what I would like for this PR

Reply:
Actually, I did not touch this one yet but you're right it should, except for the `:gitea:` case which will only show when the markdown is rendered.

Reply:
done in https://github.com/go-gitea/gitea/pull/11032/commits/5f30a0277e40d4a998e4189c611a717465c99a82.

Reply:
Also somewhat related: https://github.com/go-gitea/gitea/issues/11081

Reply:
Thanks!

and interesting for that ticket, perhaps the short code converter can see what is in the config file and not do anything if it isn't utf8mb4 (though I assume these people are just getting general errors whenever they have users try and use an emoji or import some content that has them)

Reply:
Yeah it is already affecting existing data that has emojis, it's just that they will be more prominent now with the token conversion in place.